src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/AddressDtoValidator.java

```diff
@@ -0,0 +1,69 @@
+package uk.gov.companieshouse.overseasentitiesapi.validation;
+
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.AddressDto;
+import uk.gov.companieshouse.overseasentitiesapi.validation.utils.StringValidators;
+import uk.gov.companieshouse.service.rest.err.Errors;
+
+@Component
+public class AddressDtoValidator {
+
+    public Errors validate(String field, AddressDto addressDto, Errors errs, String loggingContext) {
+        validatePropertyNameNumber(field, addressDto.getPropertyNameNumber(), errs, loggingContext);
+        validateLine1(field, addressDto.getLine1(), errs, loggingContext);
+        validateLine2(field, addressDto.getLine2(), errs, loggingContext);
+        validateTown(field, addressDto.getTown(), errs, loggingContext);
+        validateCounty(field, addressDto.getCounty(), errs, loggingContext);
+        validateCountry(field, addressDto.getCountry(), errs, loggingContext);
+        validatePostcode(field, addressDto.getPostcode(), errs, loggingContext);
+        return errs;
+    }
+
+    private boolean validatePropertyNameNumber(String field, String propertyNameNumber, Errors errs, String loggingContext) {
+        String location = getAddressFieldName(field, AddressDto.PROPERTY_NAME_NUMBER_FIELD);
+        return StringValidators.validateStringNotBlank(propertyNameNumber, location, errs, loggingContext)
+                && StringValidators.validateLength(propertyNameNumber, 50, location, errs, loggingContext)
```

Comment:
Guess I missed the previous review on this, but... 'validateLength()' - is this checking max length or min length?. Would be nice to reflect which it is in the method name.

Reply:
Updated to max length.